contracts/zero-ex/contracts/src/fixins/FixinProtocolFees.sol

```diff
@@ -0,0 +1,116 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "../external/FeeCollector.sol";
+import "../features/libs/LibTokenSpender.sol";
+
+/// @dev Helpers for collecting protocol fees.
+abstract contract FixinProtocolFees {
+    bytes32 immutable feeCollectorCodeHash;
+
+    constructor() internal {
+        feeCollectorCodeHash = keccak256(type(FeeCollector).creationCode);
+    }
+
+    /// @dev   Collect the specified protocol fee in either WETH or ETH. The
+    ///        fee is stored in a per-pool fee collector contract.
+    /// @param poolId The pool ID for which a fee is being collected.
+    /// @param amount The amount of ETH/WETH to be collected.
+    /// @param weth The WETH token contract.
+    function _collectProtocolFee(
+        bytes32 poolId,
+        uint256 amount,
+        IERC20TokenV06 weth
+    )
+        internal
+    {
+        require(
+            msg.value == amount || msg.value == 0,
+            "FixinProtocolFees/WRONG_ETHER_AMOUNT");
```

Comment:
Hmm, I wonder if we'll ever allow filling a WETH order with ETH, causing this to break. Guess we can cross that bridge when we get there.